pkg/apis/componentconfig/v1alpha1/defaults.go

```diff
@@ -45,7 +45,7 @@ func SetDefaults_CloudControllerManagerConfiguration(obj *CloudControllerManager
 
 func SetDefaults_KubeControllerManagerConfiguration(obj *KubeControllerManagerConfiguration) {
 	zero := metav1.Duration{}
-	if len(obj.Controllers) == 0 {
+	if obj.Controllers == nil || len(obj.Controllers) == 0 {
```

Comment:
a nil slice is also zero-length... this shouldn't be needed

Reply:
ok, right, removing then